lib/util.ex

```diff
@@ -75,24 +70,51 @@ defmodule Util do
 
     case HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers) do
       {:ok, resp} ->
-        resp.body
-        |> JSON.decode
-        |> format_decoded_response
+        tourney_map = resp.body
+                      |> JSON.decode
+                      |> get_tourney_map
+
+        total_pages = tourney_map
+                      |> Map.get("pageInfo")
+                      |> Map.get("totalPages")
+
+        cond do
+          total_pages < 1 ->
+            IO.puts(:stderr, "No tournament pages found.")
+            System.stop(1)
+          page_num < total_pages ->
+            # There are more pages. Add the entries from this page to the
+            # tourney list and get the next page.
+            this_page = format_decoded_response(tourney_map)
+            get_tourney_page(tourney_list ++ this_page, auth, page_num + 1)
+          page_num == total_pages ->
+            # This is the last page.
+            this_page = format_decoded_response(tourney_map)
+            tourney_list ++ this_page
+          true ->
+            # This means page_num > total_pages. This shouldn't happen, but
```

Comment:
Let's allow the CondClauseError. If we get here somehow it's probably because of some logic error during the development process, and I'd much rather see and fix the error immediately than have to spend time tracking down a bug.